tools/license/get_dep_licenses.go

```diff
@@ -46,6 +48,70 @@ const (
 )
 
 var (
+	approvedLicenses = map[string]bool{
```

Comment:
I wonder if it makes more sense to combine these maps into one map[string]string with values "approved", "reciprocal", "restricted", etc...

Reply:
Not really - the key must be the package name, so it wouldn't really help us to have the value as "approved" etc. because we also want to be able to group by the license type. So we could do map[package_name][license_type]bool, but that will actually lead to less readable code (initializing the map will be nested etc.)

Reply:
What I mean is a map from package name to license type.

And then you can still use the package name to find the corresponding license type below, and then group them. Isn't that more readable?

Reply:
Yes that's what I thought you were saying. How would I find all the "approved" licenses using your proposal? It's a linear search unless I'm missing something...

Reply:
licenses = map[string]string{
   "Apache-2.0":   "approved",
   "ISC":          "approved",
    ...
    "MPL-1.0":  "reciprocal",
    "CDDL-1.1": "reciprocal",
     ...
     "GPL-1.0-only": "restricted",
     ...
}


and then 


switch licenses {
  case "approved":
     // insert
  case "restricted":
    // insert
   ...
} 


I don't think it requires linear search. Or am I mistaken about what you are trying to do here?

Reply:
Ok, I see what you are saying. Yes, it could be done like that but it's a qualified improvement since you'd then probably also want to define & comment a string const for each license type. I think each approach has roughly equal merits...